eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/EidasSamlResource.java

```diff
@@ -69,10 +70,12 @@ public EidasSamlParserResponse post(@Valid EidasSamlParserRequest request) throw
                                 .orElseThrow();
 
         this.logAuthnRequestMdcProperties(authnRequest);
+        boolean transientPid = NameID.TRANSIENT.equals(authnRequest.getNameIDPolicy().getFormat());
```

Comment:
We don't evaluate `NameIDPolicy` until this point, so a request could be crafted without a NameID and we would throw a NPE here. You can check this out by commenting out `<saml2p:NameIDPolicy AllowCreate="true" Format="urn:oasis:names:tc:SAML:2.0:nameid-format:persistent" />` in `eidas_authn_request.xml:15` and running `EspAuthnRequestAppRuleTest`. A null check should be done here, and when null, transientPid should be false (BAU)

Reply:
Hopefully the latest commit meets your suggestion? :)